backend/src/main/java/com/bakdata/conquery/sql/execution/SqlExecutionService.java

```diff
@@ -0,0 +1,85 @@
+package com.bakdata.conquery.sql.execution;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import com.bakdata.conquery.models.error.ConqueryError;
+import com.bakdata.conquery.models.query.results.SinglelineEntityResult;
+import com.bakdata.conquery.sql.SqlQuery;
+import org.jooq.DSLContext;
+
+public class SqlExecutionService {
+
+	private final DSLContext dslContext;
+
+	public SqlExecutionService(DSLContext dslContext) {
+		this.dslContext = dslContext;
+	}
+
+	public SqlExecutionResult execute(SqlQuery sqlQuery) {
+		return dslContext.connectionResult(connection -> this.createStatementAndExecute(sqlQuery.getSqlString(), connection));
+	}
+
+	private SqlExecutionResult createStatementAndExecute(String sqlQuery, Connection connection) {
+
+		try (Statement statement = connection.createStatement();
+			 ResultSet resultSet = statement.executeQuery(sqlQuery)) {
+			int columnCount = resultSet.getMetaData().getColumnCount();
+			List<String> columnNames = this.getColumnNames(resultSet, columnCount);
+			List<SinglelineEntityResult> resultTable = this.createResultTable(resultSet, columnCount);
+
+			return new SqlExecutionResult(columnNames, resultTable);
+
+		}
+		catch (SQLException e) {
+			throw new ConqueryError.SqlError(e);
+		}
+	}
+
+	private List<SinglelineEntityResult> createResultTable(ResultSet resultSet, int columnCount) throws SQLException {
+		List<SinglelineEntityResult> resultTable = new ArrayList<>(resultSet.getFetchSize());
+		while (resultSet.next()) {
+			Object[] resultRow = this.getResultRow(resultSet, columnCount);
+			resultTable.add(new SinglelineEntityResult(resultSet.getInt(1), resultRow));
```

Comment:
hm, hier macht es dir unsere EntityResult implementation schwierig. Die entityId ist ja dictionary encoded, vmtl musst du die Klasse stärker abstrahieren zwischen SQL und Legacy

Reply:
Ja, das ist mir auch aufgefallen und geht denke ich in eine ähnliche Problematik wie mit dem IdPrinter. Da schauen wir definitiv nochmal nach, wie man das für den SQL connector anders gestalten kann